e2e-tests/common.sh

```diff
@@ -51,26 +51,99 @@ function setup() {
   http://localhost:8099/openmrs/ws/fhir2/R4/Observation/$3 2>/dev/null >>./omrs/obs.json
-  total_test_obs=$(jq '.total' ./omrs/obs.json)
+  local total_test_obs=$(jq '.total' ./omrs/obs.json)
   print_message "Total openmrs test obs ---> ${total_test_obs}"
   print_message "Counting number of patients, encounters and obs sinked to parquet files"
   total_patients_streamed=$(java -jar ./parquet-tools-1.11.1.jar rowcount ./Patient/ | awk '{print $3}')
   print_message "Total patients synced to parquet ---> ${total_patients_streamed}"
   total_encounters_streamed=$(java -jar ./parquet-tools-1.11.1.jar rowcount ./Encounter/ | awk '{print $3}')
   print_message "Total encounters synced to parquet ---> ${total_encounters_streamed}"
   total_obs_streamed=$(java -jar ./parquet-tools-1.11.1.jar rowcount ./Observation/ | awk '{print $3}')
-  print_message "Total obs synced to parquet ---> ${total_obs_streamed}"
+  print_message "Total obs synced to parquet ---> ${total_obs_streamed}"  
+  if [[ -z "$1" ]]; then
+    if [[ ${total_patients_streamed} == ${total_test_patients} && ${total_encounters_streamed} == ${total_test_encounters} && ${total_obs_streamed} == ${total_test_obs} ]] ; then
+        print_message "BATCH MODE WITH PARQUET SYNC EXECUTED SUCCESSFULLY USING ${mode} MODE"
+        cd "${HOME_PATH}"
+    else    
+        print_message "BATCH MODE WITH PARQUET SYNC TEST FAILED USING ${mode} MODE"
+        return 1
+    fi 
+  else 
+    if [[ ${total_patients_streamed} == ${total_test_patients} && ${total_encounters_streamed} == ${total_test_encounters} && ${total_obs_streamed} == ${total_test_obs} ]] ; then
+        print_message "STREAMING MODE WITH PARQUET SINK EXECUTED SUCCESSFULLY"
+    else    
+        print_message "STREAMING MODE WITH PARQUET SINK TEST FAILED"
+        return 1
+    fi
+   return 1 
+  fi        
+}
+
+###################################################################################
+# Function that validates fhir counts for batch and stream mode
+# Arguments:
+#   ?family=Clive = Argument # 1 passed to the function (positional parameter # 1)
+###################################################################################
+function validate_fhir_counts() {
+  mkdir batch_omrs
+  mkdir stream_fhir
+  mkdir batch_fhir
+  mkdir stream_omrs
+  if [[ -z "$1" ]]; then
+    print_message "Finding number of patients, encounters and obs in openmrs server"
+    curl -L -X GET -u admin:Admin123 --connect-timeout 5 --max-time 20 \
+    http://localhost:8099/openmrs/ws/fhir2/R4/Patient/ 2>/dev/null >>./batch_omrs/patients.json
+    local total_test_patients=$(jq '.total' ./batch_omrs/patients.json)
+    print_message "Total openmrs test patientss ---> ${total_test_patients}"
+    print_message "Counting number of patients, encounters and obs sinked to fhir files"
+    curl -L -X GET -u hapi:hapi --connect-timeout 5 --max-time 20 \
+    http://localhost:8098/fhir/Patient/?_summary=count 2>/dev/null >>./batch_fhir/patients.json
```

Comment:
IMO this can be possible in` validate_fhir_counts` but for `validate_parquet_counts `we can keep it the way it is coz we need to do one fhir call for each resource through which we need to switch between batch and stream yet `summary=count `cant work in the streaming case

Reply:
IMO this can be possible in` validate_fhir_counts` but for `validate_parquet_counts `we can keep it the way it is coz we need to do one fhir call for each resource through which we need to switch between batch and stream yet `summary=count `cant work in the streaming case

Reply:
I do not understand this comment. What is special about `validate_parquet_counts` that we cannot use `_summary=count`. If your concern is the query parameters, I believe you can still add `_summary=count` after query parameters. If you are saying that we are validating something beyond just count, please point out where in the code this validation is done.

If you prefer to keep it this way for now, it is fine; just add a TODO to do it later. But I want to make sure I understand what you say.